### PR TITLE
Changes for updating tutorials to work with AK 3.1

### PR DESCRIPTION
**Pull request by bbejeck**
### Description

covers the `Error Handling`  and `Naming Stateful Operations` tutorial for https://github.com/confluentinc/kafka-tutorials/issues/1256

### Staging Docs

Only the `Error Handling` tutorial had site updates so it's the only one listed, `Naming Stateful Operations` had internal changes only

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/Update_error_handling_to_latest_AK/ 
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/Update_error_handling_to_latest_AK/error-handling/kstreams.html

### CODE REVIEW

**Comment by bbejeck** [comment]
The [Error Handling Tutorial](https://confluentinc.semaphoreci.com/jobs/e4fc7cac-6db2-44d2-8395-06586b350499) passed

File: _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc [hunk 1]
```diff
@@ -65,4 +65,10 @@ Now create the following file at `src/main/java/io/confluent/developer/MaxFailur
 
 You'll add the `StreamsUncaughtExceptionHandler` to your Kafka Streams application in the next step.
 
+There's one more step you'll need to take and that is creating a custom exception class `ProcessorException` that your exception handler will process. Create this simple Java class at `src/main/java/io/confluent/developer/ProcessorException.java`
```
Comment (ybyzek) [inline _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc:68]: @bbejeck what's the tl;dr on why this change helps (whereas before not needed)?

File: _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc [hunk 2]
```diff
@@ -65,4 +65,10 @@ Now create the following file at `src/main/java/io/confluent/developer/MaxFailur
...
+
++++++
+<pre class="snippet"><code class="java">{% include_raw tutorials/error-handling/kstreams/code/src/main/java/io/confluent/developer/ProcessorException.java %}</code></pre>
++++++
+
```
Comment (ybyzek) [inline _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc:73]: @bbejeck is the note below:

1. Still accurate?
2. Still needed in the tutorial, or should it be removed?

File: _data/harnesses/naming-changelog-repartition-topics/kstreams.yml [hunk 3]
```diff
@@ -203,6 +215,12 @@ dev:
           file: tutorial-steps/dev/run-dev-app-names-with-filter.sh
           render:
             skip: true
+            
+        - name: wait for updated streams app to create new changelogs
```
Comment (ybyzek) [inline _data/harnesses/naming-changelog-repartition-topics/kstreams.yml:219]: Seems like the fix was added sleeps, good to know for potential future failures

**Review by ybyzek — APPROVED** [review]
@bbejeck Minor comment but LGTM!  Thanks for tracking these down

File: _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc (hunk 1 above)
Comment (bbejeck) [reply _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc:68]: in a nutshell, Kafka Streams now takes a look at any uncaught exception. If it's an `IllegalStateException`  or `IllegalArgumentException` (both standard exceptions in the Java API), the user won't handle those and the application shuts down.  The KT previously threw an `IllegalStateException` but that wasn't the best choice as it's a better practice to create an exception that has some business context which is the solution to fixing this KT.

File: _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc (hunk 2 above)
Comment (bbejeck) [reply _includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc:73]: Yes, it's accurate and I think we should leave it as long as the deprecated method is available.
